File: OSEUK/algorithm/week2/PGS_카펫.java
package programmers.kit.bruteforce;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Carpet {
    public static int[] solution(int brown, int yellow){
        int[] answer = new int[0];
        for (int i = 1; i <= 5000; i++) {
            for (int j = i; i * j <= 2000000; j++) {
                if (i * j == yellow && (i + 2) * (j + 2) - yellow == brown) {
                    answer =  new int[]{j + 2, i + 2};
                    break;
                }
            }
        }
        return answer;
    }
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int brown = Integer.parseInt(br.readLine());
        int yellow = Integer.parseInt(br.readLine());

        int[] result = solution(brown, yellow);
        for (int res : result) {
            System.out.print(res + " ");
        }
    }
}

File: cjh-19/algorithm/week7/PGS_카펫.js
function solution(brown, yellow) {
    var answer = [];
    // 가로 >= 세로
    let total_size = brown + yellow; // 총 개수 = 넓이 = 가로 * 세로
    let h=0, w=0;
    // 갈색 세로 길이 구하기
    for(h=3; h<=total_size/2; h++){
        if(total_size % h !== 0) continue; // 가로 길이가 정수 아니면 패스
        
        w = total_size / h; // 갈색 가로 길이
        
        // 가로 >= 세로 && 노란 격자 개수가 맞을 경우
        if(w >= h && (w-2) * (h-2) === yellow){
            answer.push(w);
            answer.push(h);
            break;
        }
    }
    
    return answer;
}